orca-core/src/test/groovy/com/netflix/spinnaker/orca/config/UserConfiguredUrlRestrictionsSpec.groovy

```diff
@@ -30,22 +35,206 @@ class UserConfiguredUrlRestrictionsSpec extends Specification {
   @Unroll
   def 'should verify allowedHostnamesRegex is set'() {
     given:
-    UserConfiguredUrlRestrictions config = new UserConfiguredUrlRestrictions.Builder()
+    UserConfiguredUrlRestrictions config = spyOn(new UserConfiguredUrlRestrictions.Builder()
         .withAllowedHostnamesRegex("")
-        .withRejectedIps([])
-        .withAllowedSchemes(['https'])
+        .build())
+
+    when:
+    config.validateURI(uri)
+
+    then:
+    thrown(IllegalArgumentException.class)
+
+    where:
+    uri << ['https://www.test.com', 'https://201.152.178.212', 'https://www.test_underscore.com']
+  }
+
+  @Unroll
+  def 'should exclude common internal URL schemes by default'() {
+    given:
+    UserConfiguredUrlRestrictions config = spyOn(new UserConfiguredUrlRestrictions.Builder().build())
+
+    when:
+    config.validateURI(uri)
+
+    then:
+    thrown(IllegalArgumentException.class)
+
+    where:
+    uri << [
+        'https://orca',
+        'https://clouddriver.svc.cluster.local',
+        'https://echo.internal',
+        'https://orca.spinnaker',
```

Comment:
Sould check on https://spin-orca/ as I believe that's a very common one of the patterns... just on principle (looks like the regex SHOULD catch it).

Reply:
I've added `spin-orca` to that test